TypeCobol.Analysis/Graph/ControlFlowGraph.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace TypeCobol.Analysis.Graph
+{
+    /// <summary>
+    /// A set of basic blocks, each of which has a list of successor blocks and some other information.
+    /// Each block consists of a list of instructions, each of which can point to previous instructions that compute the operands it consumes.
+    /// </summary>
+    /// <typeparam name="N"></typeparam>
+    /// <typeparam name="D"></typeparam>
+    public class ControlFlowGraph<N, D>
+    {
+        /// <summary>
+        /// BasicBlock callback type.
+        /// </summary>
+        /// <param name="block">The BasicBlock</param>
+        /// <param name="cfg">The Control Flow Graph in which the basic Block belongs to.</param>
+        /// <returns>true if ok, false otherwise</returns>
+        public delegate bool BasicBlockCallback(BasicBlock<N, D> block, ControlFlowGraph<N, D> cfg);
+
+        /// <summary>
+        /// Flag on a Cfg.
+        /// </summary>
+        [Flags]
+        public enum Flags : uint
+        {
+            Compound = 0x01 << 0,       //Flag to indicate that this Cfg graph has subgraphs.
```

Comment:
Turn this unique flag into a boolean property `HasSubgraphs`.

Reply:
I think we can keep Flags enum, even if there is only 1 value.
Has we already have a flag mechanism in a lot of class, it will be consistent.
And it'll be there if we need to add other values.

Reply:
Sure